extensions/azurecore/src/account-provider/auths/azureAuth.ts

```diff
@@ -344,6 +435,38 @@ export abstract class AzureAuth implements vscode.Disposable {
 	}
 
 
+	public async getTenantsMsal(token: string): Promise<Tenant[]> {
+		interface TenantResponse { // https://docs.microsoft.com/en-us/rest/api/resources/tenants/list
```

Comment:
You should put these in a common location with other interfaces (we have a couple in azurecore already, pick one that makes the most sense to you, even if it's in this file. But it shouldn't generally be inline like this since that just adds clutter to the method and makes it difficult to reuse)